sdcm/logcollector.py

```diff
@@ -0,0 +1,819 @@
+                                                      global_ip=instance['PublicIpAddress']))
+
+    def get_gce_instances_by_testid(self):
+        instances = list_instances_gce({"TestId": self.test_id}, running=True)
+        for instance in instances:
+            if 'db-node' in instance.name:
+                self.db_cluster.append(CollectingNode(name=instance.name,
+                                                      ssh_login_info={
+                                                          "hostname": instance.public_ips[0],
+                                                          "user": self.params['gce_image_username'],
+                                                          "key_file": self.params['user_credentials_path']},
+                                                      instance=instance,
+                                                      global_ip=instance.public_ips[0]))
+            if 'monitor-node' in instance.name:
+                self.monitor_set.append(CollectingNode(name=instance.name,
+                                                       ssh_login_info={
+                                                           "hostname": instance.public_ips[0],
+                                                           "user": self.params['gce_image_username'],
+                                                           "key_file": self.params['user_credentials_path']},
+                                                       instance=instance,
+                                                       global_ip=instance.public_ips[0]))
+            if 'loader-node' in instance.name:
+                self.loader_set.append(CollectingNode(name=instance.name,
+                                                      ssh_login_info={
+                                                          "hostname": instance.public_ips[0],
+                                                          "user": self.params['gce_image_username'],
+                                                          "key_file": self.params['user_credentials_path']},
+                                                      instance=instance,
+                                                      global_ip=instance.public_ips[0]))
+
+    def get_running_cluster_sets_by_testid(self):
+        if self.backend == 'aws':
+            self.get_aws_instances_by_testid()
+        if self.backend == 'gce':
+            self.get_gce_instances_by_testid()
+
+    def run(self):
+        """Run collect logs process as standalone operation
+
+        Run collect log operations as standalon process or run
+        as single stage in pipeline for defined cluster sets
+        """
+        results = {}
+        if not self.test_id:
+            self.get_current_test_id()
+            LOGGER.info("Found latest test_id: {0.test_id}".format(self))
+
+        self.get_running_cluster_sets_by_testid()
+
+        local_dir_with_logs = self.get_local_testdir()
```

Comment:
if this return None, the whole thing won't work ? or do we a have a fallback here to do part of the work into a temp directory

I would love to see this working with logging into the jenkins server, i.e. collect can we can in case we are not in jenkins. i.e. db logs can be collected directly from journalctl. yes job logs and events log would be missing, but in some cases it's not that important.

Reply:
if get_local_testdir will return none, then the new dir will be created in $USER/sct_results folder
and all logs (except sct logs (critical.log, events.log, sct.logs) will be collected (db_cluster, monitor) also the test_id will be written to this new folder, and any future log collecting will stored in same folder
if run collect logs from local host, all logs will colleted from remote hosts (db, monitor, loader).
In next phase, there is an idea to find appropriate folder by test id on builders and collect sct logs.

Reply:
Can we do this a bit more explicit and not based on the username, for example if I'll run it local on my desk I just want to pass the test-id and it to work, not just when calling it on the worker.

also you'll aws credentials to run hydra commands, I'm not sure it would be available on all workers

Reply:
Fixed. Now first search on local host ( if it run manually or by builder) and then search the running instances and collect from them. For sct logs, if not found locally, will try to find on builders.